ai-platform-unified/notebooks/notebook_template.ipynb

```diff
@@ -0,0 +1,628 @@
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "sBCra4QMA2wR"
+      },
+      "source": [
+        "**If you are using Colab**, run the cell below and follow the instructions\n",
+        "when prompted to authenticate your account via oAuth.\n",
+        "\n",
+        "**Otherwise**, follow these steps:\n",
+        "\n",
+        "1. In the GCP Console, go to the [**Create service account key**\n",
+        "   page](https://console.cloud.google.com/apis/credentials/serviceaccountkey).\n",
+        "\n",
+        "2. Click **Create service account**.\n",
+        "\n",
+        "3. In the **Service account name** field, enter a name, and\n",
+        "   click **Create**.\n",
+        "\n",
+        "4. In the **Grant this service account access to project** section, click the **Role** drop-down list. Type \"AI Platform\"\n",
+        "into the filter box, and select\n",
+        "   **AI Platform Administrator**. Type \"Storage Object Admin\" into the filter box, and select **Storage Object Admin**.\n",
+        "\n",
+        "5. Click *Create*. A JSON file that contains your key downloads to your\n",
+        "local environment.\n",
+        "\n",
+        "6. Enter the path to your service account key as the\n",
+        "`GOOGLE_APPLICATION_CREDENTIALS` variable in the cell below and run the cell."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "PyQmSRbKA8r-"
+      },
+      "source": [
+        "import sys\n",
+        "import os\n",
+        "\n",
+        "# If you are running this notebook in Colab, run this cell and follow the\n",
+        "# instructions to authenticate your GCP account. This provides access to your\n",
+        "# Cloud Storage bucket and lets you submit training jobs and prediction\n",
+        "# requests.\n",
+        "\n",
+        "# If on AI Platform, then don't execute this code\n",
+        "    if not os.path.exists('/opt/deeplearning/metadata/env_version'):\n",
+        "      if 'google.colab' in sys.modules:\n",
+        "        from google.colab import auth as google_auth\n",
+        "        google_auth.authenticate_user()\n",
+        "\n",
+        "      # If you are running this notebook locally, replace the string below with the\n",
+        "      # path to your service account key and run this cell to authenticate your GCP\n",
+        "      # account.\n",
+        "      else:\n",
+        "        %env GOOGLE_APPLICATION_CREDENTIALS ''"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "zgPO1eR3CYjk"
+      },
+      "source": [
+        "### Create a Cloud Storage bucket\n",
+        "\n",
+        "**The following steps are required, regardless of your notebook environment.**\n",
+        "\n",
+        "When you submit a training job using the Cloud SDK, you upload a Python package\n",
```

Comment:
This part, if it's meant to be part of the 'template' text, is phrased too specifically -- not all the examples that are required to use this template will directly show a training job. E.g., some MP examples that show DSL control structure, FS examples, etc. (A GCS bucket is almost always required for some purpose even so, though I can think of a few exceptions).

Reply:
Okay, I'm adding a TODO to explain why your tutorial uses the Cloud Storage bucket, and leaving in the existing wording as an example.

Reply:
Added.

Reply:
thanks, that change lgtm.